datalad/core/distributed/clone.py

```diff
@@ -410,6 +410,33 @@ def clone_dataset(
 
     dest_path = destds.pathobj
 
+    # check for configured URL mappings, either in the given config manager
+    # or in the one of the destination dataset, which is typically not existent
+    # yet and the process config manager is then used effectively
+    url_mapping = ensure_list((cfg or destds.config).get(
+        'datalad.url.substitute', default=[], get_all=True))
+    if url_mapping:
+        _srcs = []
+        # decode the rule specifications
+        mspecs = get_replacement_dict(url_mapping)
+        # we process the candidate in order to maintain any prioritization
+        # encoded in it (e.g. _get_flexible_source_candidates_for_submodule)
+        # if we have a matching mapping replace the URL in its position
+        for s in srcs:
+            for match_ex, subst_ex in mspecs.items():
+                # try to map, would return unchanged, if there is no match
+                mapped = re.sub(match_ex, subst_ex, s)
+                if mapped != s:
+                    _srcs.append(mapped)
+                    # MIH: unclear if readding the original URL as a fallback,
+                    # in case the mapping was stupid is desirable.
+                    # if the candidate URL was hopeless to begin with, it would
+                    # result in two failed attempts instead of one
```

Comment:
Re comment - I agree. Don't "fix" explicitly given configs. However, we could use the hint mechanism to point out, that a candidate failed that was the result of such a replacement.

Reply:
To be clear: By agreeing I mean: Do NOT add the original one.

Reply:
> Re comment - I agree. Don't "fix" explicitly given configs. However, we could use the hint mechanism to point out, that a candidate failed that was the result of such a replacement.

I am not so sure about that. Config could come from all kinds of places, also sources that are not under a user's control. If any of those substitutions is too loose, what then? File a bug and wait for a release?

Reply:
> If any of those substitutions is too loose, what then? File a bug and wait for a release?

Put a high-priority noop substitution in your confgi? ;-)
Pointing to one issue: If the config is read from everywhere (as it should), explicit priorities seem to be needed (rather than just implicitly by `git-config`-return order.